transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/internal/util/Util.java

```diff
@@ -111,14 +109,19 @@ public static boolean draggingFromValidTarget(TransformationManager manager) {
     }
 
     public static void ensureSourceFolderExists(IJavaProject javaProject,
-                                                String folder,
-                                                IProgressMonitor monitor) throws JavaModelException {
+                                                String folderName,
+                                                IProgressMonitor monitor) throws Exception {
         boolean exists = false;
         IClasspathEntry[] entries = javaProject.getRawClasspath();
-        IPath path = javaProject.getPath().append(folder);
+        IPath path = javaProject.getPath().append(folderName);
         for (IClasspathEntry entry : entries) {
             if (entry.getEntryKind() == IClasspathEntry.CPE_SOURCE && entry.getPath().equals(path)) {
                 exists = true;
+                File folder = javaProject.getResource().getLocation().append(folderName).toFile();
+                if (!folder.exists()) {
+                    if (!folder.mkdirs()) throw new Exception(Messages.bind(Messages.Util_UnableToCreateSourceFolder, folder));
```

Comment:
Regarding the exception, unless we're going to handle the exception differently, I see no point to being specific about what we use.  There are already more than one exception thrown by the method, all of which indicate a fatal situation that gets logged, displayed, and prevents to user from proceeding (In the editor, they all get generically wrapped in a PartInitException anyways).  If we ever have some need to make a delineation, I can change this.

Regarding the new line, that's really a style, and therefore, religious :-), view. I prefer single-line statements when possible for readability, code brevity, and the slight increased editing productivity.